### DISCUSSION TITLE
response_model does not support __init_subclass__ parameters

### DISCUSSION THREAD

**Discussion by nayaverdier** [fastapi/fastapi#8714] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import APIRouter
from pydantic import BaseModel

class BaseResp(BaseModel):
    def __init_subclass__(self, parameter: str, **kwargs):
        super().__init_subclass__(**kwargs)
        self.parameter = parameter

class RespFoo(BaseResp, parameter="foo"):
    foo: str

router = APIRouter()

@router.get("/", response_model=RespFoo)
def root():
    return {}
```

### Description

Running this code, the following error appears:
```
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File ".../example.py", line 19, in <module>
    def root():
  File ".../python3.10/site-packages/fastapi/routing.py", line 582, in decorator
    self.add_api_route(
  File ".../python3.10/site-packages/fastapi/routing.py", line 525, in add_api_route
    route = route_class(
  File ".../python3.10/site-packages/fastapi/routing.py", line 363, in __init__
    ] = create_cloned_field(self.response_field)
  File ".../python3.10/site-packages/fastapi/utils.py", line 88, in create_cloned_field
    use_type = create_model(original_type.__name__, __base__=original_type)
  File ".../python3.10/site-packages/pydantic/main.py", line 990, in create_model
    return type(__model_name, (__base__,), namespace)
  File ".../python3.10/site-packages/pydantic/main.py", line 367, in __new__
    cls = super().__new__(mcs, name, bases, new_namespace, **kwargs)
  File "/usr/lib/python3.10/abc.py", line 106, in __new__
    cls = super().__new__(mcls, name, bases, namespace, **kwargs)
TypeError: BaseResp.__init_subclass__() missing 1 required positional argument: 'parameter'
```

### Operating System

Linux

### FastAPI Version

0.70.1

### Python Version

3.10.1

**Comment by harunyasar** [comment]
This is not a thing that `response_model` is supposed to support explicitly because it is a Python feature. `BaseResp` is not a base class of your model in fact it is a sub-class of `BaseModel`. So, you could do as follows.

```python
class BaseResp(BaseModel):
    def __init_subclass__(cls, parameter: str = "default", **kwargs):
        super().__init_subclass__(**kwargs)
        cls.parameter = parameter
```
